pkg/kapp/preflight/registry.go

```diff
@@ -0,0 +1,87 @@
+package preflight
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/pflag"
+	ctldgraph "github.com/vmware-tanzu/carvel-kapp/pkg/kapp/diffgraph"
+)
+
+const preflightFlag = "preflight"
+
+type Registry struct {
+	known map[string]Check
+}
+
+func NewRegistry(checks map[string]Check) *Registry {
+	registry := &Registry{}
+	for name, check := range checks {
+		registry.AddCheck(name, check)
+	}
+	return registry
+}
+
+func (c *Registry) String() string {
+	return fmt.Sprintf("%+v", c.known)
+}
+
+func (c *Registry) Type() string {
+	return fmt.Sprintf("%T", c)
+}
+
+func (c *Registry) Set(s string) error {
+	if c.known == nil {
+		return errors.New("no preflight checks exist")
+	}
+
+	mappings := strings.Split(s, ",")
+	for _, mapping := range mappings {
+		set := strings.SplitN(mapping, "=", 2)
+		if len(set) != 2 {
+			return fmt.Errorf("unable to parse check definition %q, too many '='. Must follow the format check={true||false}", mapping)
+		}
+		key, value := set[0], set[1]
+
+		if _, ok := c.known[key]; !ok {
+			return fmt.Errorf("unknown preflight check %q specified", key)
+		}
+
+		enabled, err := strconv.ParseBool(value)
+		if err != nil {
+			return fmt.Errorf("unable to parse boolean representation of %q: %w", mapping, err)
+		}
+		c.known[key].SetEnabled(enabled)
+	}
+	return nil
+}
+
+func (c *Registry) AddFlags(flags *pflag.FlagSet) {
+	knownChecks := []string{}
+	for name, _ := range c.known {
+		knownChecks = append(knownChecks, name)
+	}
+	flags.Var(c, preflightFlag, fmt.Sprintf("preflight checks to run. Available preflight checks are %s", strings.Join(knownChecks, ",")))
```

Comment:
Maybe? This tries to keep it cleaner by only doing a slice of the check names in the usage method. If there is concern with include the list of defaults I can remove it. I personally err on the side of more information rather than less though. IMO it should be self documenting and the current implementation should adhere to that in as succinct a way as possible.

Reply:
My main concern was that the help section for preflight flag would grow to be very long as we add more preflight checks, but it looks fine as of now, so we can think about it later.